analytics-datastore-clickhouse/swarm.sh

```diff
@@ -16,92 +16,69 @@ readonly ROOT_PATH
+    docker::deploy_config_importer "$COMPOSE_FILE_PATH/importer" "clickhouse-config-importer" "clickhouse"
 
-    config::remove_config_importer clickhouse-config-importer
-
-    # Ensure config importer is removed
-    config::await_service_removed instant_clickhouse-config-importer
+  ################################ DOWN ################################
+  elif [[ "${ACTION}" == "down" ]]; then
+    log info "Scaling down Analytics Datastore Clickhouse"
 
-    log info "Removing stale configs..."
-    config::remove_stale_service_configs "$COMPOSE_FILE_PATH"/importer/docker-compose.config.yml "clickhouse"
+    for service_name in "${service_names[@]}"; do
+      try "docker service scale instant_$service_name=0" "Failed to scale down $service_name"
+    done
 
-    if [[ "${STATEFUL_NODES}" == "cluster" ]]; then
-      docker::deploy_sanity analytics-datastore-clickhouse-01 analytics-datastore-clickhouse-02 analytics-datastore-clickhouse-03 analytics-datastore-clickhouse-04
-    else
-      docker::deploy_sanity analytics-datastore-clickhouse
-    fi
-  elif [[ "${ACTION}" == "down" ]]; then
-    if [[ "${STATEFUL_NODES}" == "cluster" ]]; then
-      try "docker service scale instant_analytics-datastore-clickhouse-01=0" "Failed to scale down analytics-datastore-clickhouse-01"
-      try "docker service scale instant_analytics-datastore-clickhouse-02=0" "Failed to scale down analytics-datastore-clickhouse-02"
-      try "docker service scale instant_analytics-datastore-clickhouse-03=0" "Failed to scale down analytics-datastore-clickhouse-03"
-      try "docker service scale instant_analytics-datastore-clickhouse-04=0" "Failed to scale down analytics-datastore-clickhouse-04"
-    else
-      try "docker service scale instant_analytics-datastore-clickhouse=0" "Failed to scale down analytics-datastore-clickhouse"
-    fi
+  ################################ DESTROY ################################
   elif [[ "${ACTION}" == "destroy" ]]; then
     docker::service_destroy clickhouse-config-importer
 
-    if [[ "${STATEFUL_NODES}" == "cluster" ]]; then
-      docker::service_destroy analytics-datastore-clickhouse-01
-      docker::service_destroy analytics-datastore-clickhouse-02
-      docker::service_destroy analytics-datastore-clickhouse-03
-      docker::service_destroy analytics-datastore-clickhouse-04
+    for service_name in "${service_names[@]}"; do
+      docker::service_destroy "$service_name"
+    done
 
-      docker::try_remove_volume clickhouse-data-01
-      docker::try_remove_volume clickhouse-data-04
+    if [[ "$NODE_MODE" == "cluster" ]]; then
+      docker::volume_destroy clickhouse-data-01 clickhouse-data-04
+      log warn "Volumes are only deleted on the host on which the command is run. Cluster volumes on other nodes are not deleted"
     else
-      docker::service_destroy analytics-datastore-clickhouse
-      docker::try_remove_volume clickhouse-data
+      docker::volume_destroy clickhouse-data
```

Comment:
Call to non-existent function.

Reply:
ops nice catch! fixed in [3c9156e](https://github.com/jembi/platform/pull/188/commits/3c9156ed2509979f675fa477ed694b0df73c8631)